doc/guides/modules.rst

```diff
@@ -0,0 +1,198 @@
+
+.. _modules:
+
+Modules (External projects)
+############################
+
+Zephyr relies on the source code of several externally maintained projects in
+order to avoid reinventing the wheel and to reuse as much well-established,
+mature code as possible when it makes sense. In the context of Zephyr's build
+system those are called *modules*. These modules must be integrated with the
+Zephyr build system, which is described in more detail in other sections on
+this page.
+
+There are several categories of external projects that Zephyr depends on,
+including:
+
+- Debugger integration
+- Silicon vendor Hardware Abstraction Layers (HALs)
+- Cryptography libraries
+- File Systems
+- Inter-Process Communication (IPC) libraries
+
+The build system variable :makevar:`ZEPHYR_MODULES` is a `CMake list`_ of
+absolute paths to the directories containing Zephyr modules. These modules
+contain :file:`CMakeLists.txt` and :file:`Kconfig` files describing how to
+build and configure them, respectively. Module :file:`CMakeLists.txt` files are
+added to the build using CMake's `add_subdirectory()`_ command, and the
+:file:`Kconfig` files are included in the build's Kconfig menu tree.
+
+If you have :ref:`west <west>` installed, you don't need to worry about how
+this variable is defined unless you are adding a new module. The build system
+knows how to use west to set :makevar:`ZEPHYR_MODULES`. You can add additional
+modules to this list by setting the :makevar:`ZEPHYR_EXTRA_MODULES` CMake
+variable. This can be useful if you want to keep the list of modules found with
+west and also add your own.
+
+Finally, you can also specify the list of modules yourself in various ways, or
+not use modules at all if your application doesn't need them.
+
+Topologies supported
```

Comment:
These topologies etc belong in the west document, not here; zephyr modules are a build system source code organization artifact which is orthogonal to multirepo. Not all repositories in the west manifest are zephyr modules (like net-tools).